api/http/api_deployments.go

```diff
@@ -427,13 +428,34 @@ func (d *DeploymentsApiHandlers) UploadLink(w rest.ResponseWriter, r *rest.Reque
 	d.view.RenderSuccessGet(w, link)
 }
 
+const maxMetadataSize = 2048
+
 func (d *DeploymentsApiHandlers) CompleteUpload(w rest.ResponseWriter, r *rest.Request) {
 	ctx := r.Context()
 	l := log.FromContext(ctx)
 
 	artifactID := r.PathParam(ParamID)
 
-	err := d.app.CompleteUpload(ctx, artifactID, d.config.EnableDirectUploadSkipVerify)
+	var metadata *model.DirectUploadMetadata
+	if d.config.EnableDirectUploadSkipVerify {
+		var directMetadata model.DirectUploadMetadata
+		bodyBuffer := make([]byte, maxMetadataSize)
+		n, err := io.ReadFull(r.Body, bodyBuffer)
+		r.Body.Close()
+		if err != nil && err != io.EOF && err != io.ErrUnexpectedEOF {
+			l.Errorf("error reading post body data: %s (read: %d)", err.Error(), n)
+		}
+		err = json.Unmarshal(bodyBuffer[:n], &directMetadata)
```

Comment:
If `err != nil`, you don't want to continue with the Unmarshal, right?
Maybe you should stuff this code in an `else`.

Reply:
fixed.